### PR TITLE
Added dynamic hrefs through property paths. This fixes #53

### LINKED ISSUE #53: make it possible to set the url from a model
Right now its only possible to define the url by hardcoding it in the annotation. However in some cases the URL needs to be fetched dynamically from the model.

```
/**
* @Rest\Relation("foo", href = ".someUrl")
*/
```

### PR DESCRIPTION
**Pull request by baldurrensch**

### CODE REVIEW

**Comment by lsmith77** [comment]
that was fast :)

**Comment by baldurrensch** [comment]
That's how we roll. :)

File: Factory/LinkFactory.php [hunk 1]
```diff
@@ -86,6 +87,15 @@ public function createLinkFromMetadata(RelationMetadataInterface $relationMetada
     {
         if (null !== $relationMetadata->getUrl()) {
             $href = $relationMetadata->getUrl();
+
+            // Check to see if we can get a value through a property path
+            try {
+                $propertyPath = new PropertyPath(trim($href, "."));
+                $href = $this->propertyAccessor->getValue($object, $propertyPath);
+            } catch (NoSuchPropertyException $e) {
+                // Don't do anything in this case. There is unfortunately no way to check the
```
Comment (adrienbrault) [inline Factory/LinkFactory.php]: The ParametersFactory currently only support values starting by `.` or `[`. https://github.com/TheFootballSocialClub/FSCHateoasBundle/blob/master/Factory/ParametersFactory.php#L28

**Comment by adrienbrault** [comment]
thanks!

**Pull request merged**